crates/service_utils/src/helpers.rs

```diff
@@ -401,3 +411,91 @@ pub fn parse_config_tags(
         }
     }
 }
+
+pub async fn execute_webhook_call<T>(
+    webhook_object: Value,
+    payload: T,
+    config_version_opt: &Option<String>,
+    event: WebhookEvent,
+) -> Result<(), AppError>
+where
+    T: Serialize,
+{
+    match serde_json::from_value::<Webhook>(webhook_object) {
+        Ok(webhook) => {
+            let http_client = reqwest::Client::new();
+            let mut header_array = webhook
+                .headers
+                .into_iter()
+                .filter_map(|key| match key {
```

Comment:
why do we need to do a filter operation over here ?
this is counter intuitive, `webhook.headers`, is of type `Vec<HeadersEnum>`, so it does not make sense to filter here

to avoid repetition of keys in the headers (which headers anyways disregards and keeps only one entry as its a map), you can simply add if checks for each value present in the `HeadersEnum` which makes the handling of each header explicit, as it is supposed to be different behaviour for each

or this could have simply been a `map`

Reply:
But config_version_id is optional, so filter_map

Reply:
How please explain